terraform/ec2/win/main.tf

```diff
@@ -72,18 +73,20 @@ resource "aws_instance" "cwagent" {
   tags = {
     Name = "cwagent-integ-test-ec2-windows-${var.test_name}-${module.common.testing_id}"
   }
+  depends_on = [aws_ssm_parameter.upload_ssm]
 }
 
 # Size of windows json is too large thus can't use standard tier
 resource "aws_ssm_parameter" "upload_ssm" {
-  count = var.use_ssm == true && length(regexall("/feature/windows", var.test_dir)) > 0 ? 1 : 0
+  count = length(regexall("/feature/windows", var.test_dir)) > 0 ? 1 : 0
```

Comment:
We can upload the value to ssm for all feature tests since it will be deleted by terraform destroy anyway. This will allow us to use ssm for userdata test.